### PR TITLE
Make sure payment template is available during coin selection and fee estimation is using it for multisig

### PR DESCRIPTION
**Pull request by paweljakubas**
- [x] Added script template of native script to tx skeleton and update tx size estimation basing on ledger's cddl

### Comments

Minting/delegation/validity for multisig is going to be addressed in coming PRs. At this point single/multiple output multisig txs along with metadata support should work. 

### Issue Number
adp-2221

### CODE REVIEW

File: lib/wallet/src/Cardano/Wallet/Shelley/Transaction.hs [hunk 1]
```diff
@@ -1820,9 +1826,10 @@ estimateTxSize era skeleton =
 
     -- Total number of signatures the scripts require
     numberOf_ScriptVkeyWitnesses
-        = sumVia scriptRequiredKeySigs txScripts
+        = sumVia scriptRequiredKeySigs txMintOrBurnScripts
+        + maybe 0 scriptRequiredKeySigs txPaymentTemplate
```
Comment (Anviking) [inline lib/wallet/src/Cardano/Wallet/Shelley/Transaction.hs]: The effect of this line seems to be to: "add `sizeOf_VKeyWitness * scriptRequiredKeySigs` if a payment template exists." 

You also seem to be adding `determinePaymentTemplateSize` _once_.

I'd guess it should rather be `determinePaymentTemplateSize * nInputs * scriptRequiredKeySigs`?

**Review by Anviking — APPROVED** [review]
Don't see how the details work out but high level lgtm 👍

File: lib/wallet/src/Cardano/Wallet/Shelley/Transaction.hs (hunk 1 above)
Comment (paweljakubas) [reply lib/wallet/src/Cardano/Wallet/Shelley/Transaction.hs]: indeed! Thanks Johannes!

**Comment by paweljakubas** [comment]
bors try

**Review by piotr-iohk — APPROVED** [review]
It appears that with most recent commit [7402995](https://github.com/input-output-hk/cardano-wallet/pull/3502/commits/7402995c50ef1608b50f5c135a4dafcb741393b0) this fixes ADP-2221. So far I haven't seen "invalid_transaction" on submission on preview network.

Also enabled submitting tx + additional assertions on following txs from shared wallets (e2e tests):
 - single output
 - multi output
 - multi-assets
 - validity intervals

**Pull request merged**